generator/ProtocolTestsGenerator/smithy-dotnet-codegen/src/main/java/software/amazon/smithy/dotnet/codegen/DotnetImportContainer.java

```diff
@@ -0,0 +1,116 @@
+    );
+    private static final List<String> coreImports = Arrays.asList(
+            "Amazon.Runtime",
+            "Amazon.Runtime.Internal",
+            "Amazon.Runtime.Internal.Transform"
+    );
+    private static final List<String> protocolTestImports = Arrays.asList(
+            "AWSSDK.ProtocolTests",
+            "Microsoft.VisualStudio.TestTools.UnitTesting",
+            "Amazon.Runtime.Internal.Transform",
+            "AWSSDK_DotNet35.UnitTests.TestTools"
+    );
+    private final Map<String, Map<String,String>> imports = new TreeMap<>();
+    private final DotnetSettings settings;
+    private final String localNamespace;
+    public DotnetImportContainer(String namespace, DotnetSettings settings) {
+        this.settings = settings;
+        this.localNamespace = namespace;
+
+    }
+
+    @Override
+    public void importSymbol(Symbol symbol, String alias) {
+        var symbolNamespace = symbol.getNamespace();
+
+        // Only import symbols in other namespaces.
+        if (!symbolNamespace.equals(localNamespace)) {
+            var namespaceImports = imports.computeIfAbsent(symbolNamespace, ns -> new TreeMap<>());
+            namespaceImports.put(alias, symbol.getName());
+        }
+    }
+
+    public ImportContainer addImport(String namespace, String name) {
+        return addImport(namespace,name,name);
+    }
+    public ImportContainer addImport(String namespace, String name, String alias) {
+        Map<String,String> namespaceImports = imports.computeIfAbsent(namespace, ns -> new TreeMap<>());
+        namespaceImports.put(name, alias);
+        return this;
+    }
+
+    @Override
+    public String toString(){
+        if(imports.isEmpty()){
+            return "";
+        }
+        var builder = new StringBuilder();
+        for(var entry : imports.entrySet()){
+            String formattedUsingStatements = formatNamespaceEntries(entry.getKey(),entry.getValue());
+            builder.append(formattedUsingStatements);
+        }
+        builder.append("\n");
+        return builder.toString();
+    }
+
+    public String formatNamespaceEntries(String namespace, Map<String,String> namespaceImports){
+        var builder = new StringBuilder();
+        for(var entry : namespaceImports.entrySet()){
+            builder.append("using ").append(entry.getKey()).append(";\n");
+        }
+        return builder.toString();
+    }
+
+    public void addSystemImport(String namespace) {
+        for(String systemImport: systemImports){
+            addImport(namespace, systemImport);
+        }
+    }
+
+    public void addCoreImport(String namespace) {
+        for(String coreImport: coreImports){
+            addImport(namespace, coreImport);
+        }
+    }
+
+    public void addMarshallImports(String namespace, String service) {
+        addImport(namespace,service);
+        addImport(namespace, service + ".Model");
+        addImport(namespace, service + ".Model.Internal.MarshallTransformations");
```

Comment:
I decided to put this in here because in the .NET SDK all marshallers/unmarshallers live in the `.Model.Internal.MarshallTransformations` namespace. So I thought this is something we could re-use for a full smithy codegenerator later. i.e. make it generic and not specific to protocol tests.